### DISCUSSION TITLE
Difficulty upgrading TC_PAIRS workflow from METplus 4 to METplus 5

### DISCUSSION THREAD

**Discussion by mollybsmith-noaa** [dtcenter/METplus#2216] [Configuration] [answered]
Hi all,

I'm trying to upgrade my TC_PAIRS workflow to METplus 5.0.1, in advance of the 5.1.0 release. I'm starting by using the same configuration file that I had with METplus 4.1.1, but with this configuration, TC_PAIRS runs for one init time and then stops. I don't see anything obvious in the documentation about syntax changes between v4 and v5, so do you have any idea what may be causing this?

Specifics:

I’ve run two TC_Pairs verifications on Jet for NHC forecasts initialized 2023-06-01-18. They should be identical in terms of configuration except that one is running METplus 5 / MET 11 and one is running METplus 4 / MET 10.

Example output file for METplus 4: `/lfs1/BMC/amb-verif/tcmet/output/metplus_4_test/tc_pairs_post/nhc/GFSO/2023060818/tc_pairs_al02.dat.tcst`
Example output file for METplus 5: `/lfs1/BMC/amb-verif/tcmet/output/metplus_5_test/tc_pairs_post/nhc/GFSO/2023060818/tc_pairs_al02.dat.tcst`

Note that the METplus 4 one has data, while the METplus 5 one doesn’t.

MET/METplus paths used:

  ```
 <!ENTITY MET_DIR "/contrib/met/10.1.1/met-10.1.1">
   <!ENTITY MET_BASE "/contrib/met/10.1.1/share/met">
   <!ENTITY METPLUS_PATH "/contrib/met/METplus/METplus-4.1.1">

   <!ENTITY MET_DIR "/contrib/met/11.0.1/MET-11.0.1">
   <!ENTITY MET_BASE "/contrib/met/11.0.1/share/met">
   <!ENTITY METPLUS_PATH "/contrib/met/METplus/METplus-5.0.1”>
```

Config file used for both runs:

`/home/role.amb-verif/tcmet/config/metplus/TCPairs_nhc_jet_rt_post_metplus_5.conf
`

METplus final config files:

METplus 4: `/lfs1/BMC/amb-verif/tcmet/output/metplus_4_test/tc_pairs_post/nhc/metplus_final.conf`
METplus 5: `/lfs1/BMC/amb-verif/tcmet/output/metplus_5_test/tc_pairs_post/nhc/metplus_final.conf.20230614175940`

METplus verbose logs:

METplus 4: `/lfs1/BMC/amb-verif/tcmet/output/metplus_4_test/tc_pairs_post/nhc/logs/metplus.log.20230614162552`
METplus 5: `/lfs1/BMC/amb-verif/tcmet/output/metplus_5_test/tc_pairs_post/nhc/logs/metplus.log.20230614175940`

Let me know if you need further information, and thanks for looking into this!

**Comment by georgemccabe** [comment, marked as answer]
This is a change in behavior moving from v4.1 to v5.0 that only applies to TCPairs. To fix this, remove `LOOP_ORDER` from your configuration file and add `TC_PAIRS_RUN_ONCE=False`. This will process all of your initialization times.

Explanation:

The `LOOP_ORDER` METplus config variable has been deprecated.
In METplus v4.1, setting `LOOP_ORDER=times` processes each init/valid time in your configuration. TCPairs wrapper has a setting called `TC_PAIRS_RUN_ONCE` that only runs the first init/valid time and is set to True by default. This setting was ignored prior to v5.0.0 when `LOOP_ORDER=times` was set, which was quite confusing to a user. With the removal of `LOOP_ORDER`, the setting is never ignored, but in result the behavior may change when upgrading METplus versions.

I am adding a check if `LOOP_ORDER=times` and `TC_PAIRS_RUN_ONCE` is not set to output a warning message and force TC_PAIRS_RUN_ONCE=False to preserve the old behavior. I also added a debug log message to alert the user if the use case is configured to only process the first run time and how to change that if desired. This will be available in METplus v5.1.0.

**Comment by mollybsmith-noaa** [comment]
This does work, thanks! The debug message also sounds like a good idea.
